enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx

```diff
@@ -56,60 +55,63 @@ export const _InteractiveQuestion = ({
   useEffect(() => {
     const fetchQBData = async () => {
       const { location, params } = getQuestionParameters(questionId);
-      dispatch(initializeQB(location, params));
+      try {
+        await dispatch(initializeQBRaw(location, params));
+      } catch (error) {
+        setLoading(false);
+      }
     };
 
-    fetchQBData().then(() => setLoading(false));
+    fetchQBData();
   }, [dispatch, questionId]);
 
-  if (!loading && !queryResults) {
+  useEffect(() => {
+    if (queryResults) {
+      setLoading(false);
+    }
+  }, [queryResults]);
+
+  if (loading) {
```

Comment:
We'll either need to create a new `LoadingAndErrorWrapper` to use SDK components or do something like this. I'll think about ways we can do this without needing to repeat ourselves while still having control and type safety